plugin/clang.vim

```diff
@@ -941,16 +953,25 @@ func! s:ClangExecute(root, clang_options, line, col)
     " Please note that '--remote-expr' executes expressions in server, but
     " '--remote-send' only sends keys, which is same as type keys in server...
     " Here occurs a bug if uses '--remote-send', the 'col(".")' is not right.
-    let l:keys = printf('ClangExecuteDone("%s","%s")', l:tmps[0], l:tmps[1])
-    let l:vcmd = printf('%s -s --noplugin --servername %s --remote-expr %s',
-                      \ g:clang_vim_exec, shellescape(v:servername), shellescape(l:keys))
-    let l:command = '('.l:command.';'.l:vcmd.') &'
-    call s:PDebug("s:ClangExecute::cmd", l:command, 2)
-    call system(l:command, l:src)
+    if s:is_windows
+      let l:keys = printf("ClangExecuteDone('%s','%s')", l:tmps[0], l:tmps[1])
+      let l:vcmd = printf('%s -s --noplugin --servername %s --remote-expr %s',
+                        \ g:clang_vim_exec, shellescape(v:servername), shellescape(l:keys))
+      let l:input = tempname()
+      call writefile(split(l:src, "\n"), l:input)
+      let l:command = 'type '.l:input.' | '.l:command.' && '.l:vcmd
```

Comment:
I think we should use `&` to combine clang and vim command. `&&` means vim command only be called when clang is successful, but `&` always calls the vim command , which is used to call the `ClangExecuteDone`. So we can display error diagnostics when have error.
```viml
let l:command = 'type '.l:input.' | '.l:command.' & '.l:vcmd
```